include/swift/AST/DiagnosticsSema.def

```diff
@@ -904,6 +904,9 @@ ERROR(nonstatic_operator_in_nominal,none,
 ERROR(nonstatic_operator_in_extension,none,
       "operator %0 declared in extension of %1 must be 'static'",
       (Identifier, TypeRepr*))
+ERROR(nonstatic_operator_in_extension_no_type,none,
+"operator %0 declared in extension must be 'static'",
+(Identifier))
```

Comment:
Maybe incorporating this into the old diagnostic using `%select` is a better idea? We can them pass a bool rather than duplicating the diagnosing code.

Reply:
Updated to use one diagnostic.